tools/autodeployment/webhook.js

```diff
@@ -48,6 +48,24 @@ function requestFilter(repo, buildType, action, ref, mainBranch) {
   );
 }
 
+/**
+ * Extract the commit for this GitHub payload, depending on the event type
+ * @param {'push' | 'release'} eventType one of 'push' or 'release'
+ * @param {Object} githubData webhook payload from GitHub, see https://github.com/Seneca-CDOT/telescope/settings/hooks
+ */
+function getSha(eventType, githubData) {
+  if (eventType === 'push') {
+    // "after": "adf299da6e6e9c3f20208ee6ab61d205a9f70eab",
+    return githubData.after;
+  }
+  if (eventType === 'release') {
+    // "target_commitish": "1a4594cfc443d307679576d9fdfa402f530a570c",
+    return githubData.release.target_commitish;
```

Comment:
How do we know that `target_commitish` is the SHA? I checked the GitHub API docs and the example shown shows the name of the `master` branch 😕

Reply:
@dbelokon https://github.com/Seneca-CDOT/telescope/issues/2650#issuecomment-1018045662

Reply:
Oh, I didn't know about that problem with the documentation. Thank you @manekenpix!